config/keyvault/config.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keyvault
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/crossplane/terrajet/pkg/config"
+
+	"github.com/crossplane-contrib/provider-jet-azure/apis/rconfig"
+	"github.com/crossplane-contrib/provider-jet-azure/config/common"
+)
+
+// getURLIDFn returns a GetIDFn that can generate Azure vault secret IDs.
+// An example identifier is as follows:
+// https://example.vault.azure.net/secrets/example-secret/c0ffee5f4d45440cb60c28672887f832
+// https://example.vault.azure.net/keys/example-key/fdf067c93bbb4b22bff4d8b7a9a56217
+func getURLIDFn(resourceType string) config.GetIDFn {
+	return func(_ context.Context, _ string, parameters map[string]interface{}, _ map[string]interface{}) (string, error) {
+		keyVaultID, err := common.ParseNameFromIDField(parameters, "key_vault_id")
+		if err != nil {
+			return "", err
+		}
+		version, err := common.GetAttributeValue(parameters, "version")
+		if err != nil { // then persistent ID is not yet available, thus do not return an error
+			return "", nil
+		}
+		name, err := common.GetAttributeValue(parameters, "name")
+		if err != nil {
+			return "", err
+		}
+
+		return fmt.Sprintf("https://%s.vault.azure.net/%s/%s/%s",
+			keyVaultID, resourceType, name, version), nil
+	}
+}
+
+// getIssuerURLIDFn is similar to getURLIDFn for vault certificate issuers and
+// does not include a version component.
+// Example:
+// https://key-vault-name.vault.azure.net/certificates/issuers/example
+// https://example-keyvault.vault.azure.net/storage/exampleStorageAcc01
+func getIssuerURLIDFn(resourceType string) config.GetIDFn {
+	return func(_ context.Context, _ string, parameters map[string]interface{}, _ map[string]interface{}) (string, error) {
+		keyVaultID, err := common.ParseNameFromIDField(parameters, "key_vault_id")
+		if err != nil {
+			return "", err
+		}
+		name, err := common.GetAttributeValue(parameters, "name")
+		if err != nil {
+			return "", err
+		}
+
+		return fmt.Sprintf("https://%s.vault.azure.net/%s/%s",
+			keyVaultID, resourceType, name), nil
+	}
+}
+
+// example ID: /subscriptions/038f2b7c-3265-43b8-8624-c9ad5da610a8/resourceGroups/alper/providers/Microsoft.KeyVault/vaults/examplekeyvault-alper/objectId/a8090ca4-b5a5-4594-aa6d-362aa682f168
+// can also contain application ID as the last component
+/*func getAccessPolicyID() config.GetIDFn {
```

Comment:
Should this be uncommented?

Reply:
I've left those commented out because I would like to configure `azurerm_key_vault_access_policy ` resource using `config.NameAsIdentifier`, which did not immediately work.